Encryption.md

```diff
@@ -19,216 +19,544 @@
+for plaintext footer mode are ‘PAR1’ to allow legacy readers to read projections of the file 
+that do not include encrypted columns.
+
+<!-- 
+ ![File Layout - Encrypted footer](doc/images/FileLayoutEncryptionPF.jpg)
+   -->
+
+ 
+### 5.4 New fields for vectorized readers
+Apache Spark and other vectorized readers slice a file by using information on the offset 
+and size of each row group. In the legacy readers, this is done by running over a list of 
+all column chunks in a row group, reading the relevant information from the column metadata, 
+adding up the size values and picking the offset of the first column as the row group offset. 
+However, vectorization needs only a row group metadata, and not the metadata of individual 
+columns. Also, in files written in an encrypted footer mode, the column metadata is not 
+available to readers without the column key. Therefore, two new fields are added to the 
+`RowGroup` structure - `file_offset` and `total_compressed_size` that are set upon file writing, 
+and allow vectorized readers to query a file even if keys to certain columns are not 
+available ('hidden columns'). Naturally, the query itself should not try to access the 
 hidden column data.
 
-## Encryption overhead
-
-The size overhead of Parquet modular encryption is negligible, since the most of the encryption 
-operations are performed on pages (the minimal unit of Parquet data storage and compression). The
-overhead order of magnitude is adding ~ 1 byte per each 10,000 bytes of original data.
-
-The throughput overhead of Parquet modular encryption depends on whether AES enciphering is done
-in software or hardware. In both cases, performing encryption on full pages (~1MB buffers) instead of
-on much smaller individual data values causes AES to work at its maximal speed. Preliminary tests
-show Parquet modular encryption throughput overhead to be up to a few percents in Java 9 workloads.
+```c
+struct RowGroup {
+...
+  /** Byte offset from beginning of file to first page (data or dictionary)
+   * in this row group **/
+  5: optional i64 file_offset
+
+  /** Total byte size of all compressed (and potentially encrypted) column data 
+   *  in this row group **/
+  6: optional i64 total_compressed_size
+}
+```
+
+## 6. Encryption Overhead
+The size overhead of Parquet modular encryption is negligible, since most of the encryption 
+operations are performed on pages (the minimal unit of Parquet data storage and compression). 
+The overhead order of magnitude is adding 1 byte per each ~30,000 bytes of original compressed 
+data.
+
+The throughput overhead of Parquet modular encryption depends on whether AES enciphering is 
+done in software or hardware. In both cases, performing encryption on full pages (~1MB buffers) 
+instead of on much smaller individual data values causes AES to work at its maximal speed. 
+Preliminary tests show Parquet modular encryption throughput overhead to be up to a few 
```

Comment:
A spec should not contain statements that will be outdated later as more data is gathered. The spec should not argue for itself to exist. It should simply and clearly state its requirements and little else.